posthog/tasks/update_cache.py

```diff
@@ -0,0 +1,44 @@
+from celery import shared_task
+from posthog.api.action import calculate_trends, get_actions
+from posthog.api.funnel import FunnelSerializer
+from posthog.models import Filter, Action, Funnel, Entity, DashboardItem
+from posthog.decorators import FUNNEL_ENDPOINT, TRENDS_ENDPOINT
+import logging
+from typing import List, Dict, Any, Union, Optional
+import json
+import datetime
+
+logger = logging.getLogger(__name__)
+
+@shared_task
+def update_cache(cache_type: str, payload: dict) -> Optional[Union[dict, List[Dict[str, Any]]]]:
+
+    if payload['dashboard_id']:
+        dashboard_item = DashboardItem.objects.filter(pk=payload['dashboard_id'])
+        if dashboard_item[0] and dashboard_item[0].deleted:
+            return None
+        dashboard_item.update(last_refresh=datetime.datetime.now())
```

Comment:
I _think_ this should be set _after_ the calculation is done. I had a case where dashboard items took a long time to load, but it said "updated: 3 minutes ago", while it was still calculating